src/compas/geometry/geometry.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from compas.data import Data
+
+
+class Geometry(Data):
+    """Base class for all geometric objects."""
+
+    def __ne__(self, other):
+        # this is not obvious to ironpython
+        return not self.__eq__(other)
+
+    def transform(self, transformation):
+        """Transform the geometry.
+
+        Parameters
+        ----------
+        transformation : :class:`Transformation`
+            The transformation used to transform the geometry.
+
+        Returns
+        -------
+        None
+
+        """
+        raise NotImplementedError
```

Comment:
this is not obvious for me :P  why is `transform` raising, but `transformed` is implemented? (i realize this is not a new change, and just been moved around, but I still don't quite get why it is like this)

Reply:
the implementation of transform is different for most objects, but transformed can just call transform (whatever it ends up being)